reVX/utilities/reeds_cols.py

```diff
@@ -0,0 +1,274 @@
+        on ``merge_col`` (column must be in the HDF5 file meta).
+    merge_col : str, optional
+        Name of column used to merge the data in the input
+        ``data_frame`` with the data in the HDF5 file. Note that this
+        column must be present in both the ``data_frame`` as well as the
+        HDF5 file meta.
+
+    Returns
+    -------
+    pandas.DataFrame
+        A pandas data frame with extra data added from input files.
+    """
+    for data_info in extra_data:
+        dsets = data_info["dsets"]
+        data_fp = data_info["data_fp"]
+
+        if data_fp.endswith(".json"):
+            with open(data_fp, "r") as fh:
+                extra_data = json.load(fh)
+        elif data_fp.endswith(".h5"):
+            with Resource(data_fp) as res:
+                extra_data = res.meta[[merge_col]]
+                for dset in dsets:
+                    extra_data[dset] = res[dset]
+                extra_data = pd.merge(data_frame[[merge_col]], extra_data,
+                                      on=merge_col)
+        else:
+            msg = ("File format not currently supported for file: {}"
+                   .format(data_fp))
+            warn(msg)
+
+        for dset in dsets:
+            data_frame[dset] = extra_data[dset]
+
+    return data_frame
+
+
+def add_reeds_columns(supply_curve_fpath, out_fp=None, lat_col="latitude",
+                      lon_col="longitude", capacity_col="capacity",
+                      extra_data=None, merge_col="sc_point_gid",
+                      filter_out_zero_capacity=True, rename_mapping=None):
+    """Add columns to supply curve required by ReEDS.
+
+    This method will add columns like "cnty_fips", "state", "county",
+    "nrel_region", "eos_mult", and "reg_mult". This method also allows
+    you to add extra columns from H5 or JSON files.
+
+    Parameters
+    ----------
+    supply_curve_fpath : str
+        Path to input supply curve. Should have standard reV supply
+        curve output columns (e.g. latitude, longitude, capacity,
+        sc_point_gid, etc.). If running from CLI, this can be a list
+        of supply curve paths.
+    out_fp : str, optional
+        Path to output file for supply curve with new columns. If
+        ``None``, the supply curve will be overwritten (i.e. the data
+        will be written to ``supply_curve_fpath``). If running from CLI,
+        this can be a list output paths (length *must* match length of
+        ``supply_curve_fpath``). By default, ``None``.
+    lat_col, lon_col : str, optional
+        Column names for latitude and longitude. These columns are
+        required to add county info. By default, ``"latitude"`` and
+        ``"longitude"``.
+    capacity_col : str, optional
+        Name of capacity column. This is used to filter out sites with
+        zero capacity, if that option is selected.
+        By default, ``"capacity"``.
+    extra_data : list of dicts
```

Comment:
Are people manually preparing this input? might be better as a dataframe / csv input with two columns.

Reply:
It would be a manual preparation, yes. The format would look something like this:

```
extra_data = [
    {
        "data_fp": "./bespoke_out.h5",
        "dsets": ["ws_mean", "capital_cost", "cf_mean-2013"]
    },
    {
        "data_fp": "./sam.json",
        "dsets": ["total_losses"]
    }
]
```
Do you have any ideas about how to format this into a CSV? I can't think of anything clean since the `dsets` input is a variable-length list.

Reply:
Gotcha, maybe not then, nevermind!